### ISSUE TITLE
change_return_type with updateOverrides/updateImplementations=false → SUCCESS + CS0508/CS0738 (no Ensure)

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#1369]
## Summary
`change_return_type` reports **SUCCESS** while leaving the solution uncompilable when `updateOverrides=false` (override return-type drift → **CS0508**) or `updateImplementations=false` (interface impl return-type drift → **CS0738**).

Peer `remove_parameter` hits the same flag-narrowing shape but **refuses** via `EnsureNoNewCompilationErrorsAsync` (CompilationError **4004**). `ChangeReturnTypeOperation` has **no Ensure** at all (file has zero Ensure helpers), so the broken rewrite is committed.

## Repro (override → CS0508)

**Before** (clean):
```csharp
namespace Demo;
public class B
{
    public virtual string M() => "a";
}
public class C : B
{
    public override string M() => "b";
}
```

Call `change_return_type` on **C.M** with `newReturnType=object`, `line` of the override, **`updateOverrides=false`**.

**After** (SUCCESS):
```csharp
public class B
{
    public virtual string M() => "a";  // unchanged
}
public class C : B
{
    public override object M() => "b";  // rewritten alone
}
```

**Actual:** build fails with `CS0508: 'C.M()': return type must be 'string' to match overridden member 'B.M()'`.

**Expected:** fail closed (CompilationError / refuse), same as `remove_parameter` Ensure — do not report SUCCESS with a broken override chain.

## Sibling (interface → CS0738)

**Before** (clean):
```csharp
namespace Demo;
public interface I
{
    string M();
}
public class C : I
{
    public string M() => "x";
}
```

Call `change_return_type` on **C.M** with `newReturnType=object`, **`updateImplementations=false`**.

**After** (SUCCESS): interface stays `string M()`; impl becomes `object M()` → `CS0738: 'C' does not implement interface member 'I.M()'. 'C.M()' cannot implement 'I.M()' because it does not have the matching return type of 'string'.`

## Root cause
- `ChangeReturnTypeOperation.ExecuteCoreAsync` (~L110–114) builds `relatedMethods` from `@params.UpdateOverrides` / `@params.UpdateImplementations`. When false, only the selected declaration is rewritten.
- `ValidateUneditableContracts` (~L103–108) only rejects **uneditable** (no-source) contracts; editable overrides/impls that are simply omitted from the rewrite are not checked for post-edit compile health.
- Commit path (~L150–167) has **no** `EnsureNoNewCompilationErrorsAsync`.
- Contrast: `RemoveParameterOperation` (~L138, ~L647–684) runs Ensure after rewrite and blocks the analogous `UpdateOverrides=false` / `UpdateImplementations=false` cases with **4004**.
- Same missing-Ensure class as `reorder_parameters` (**#1367**) and `add_parameter` (**#1368**); this is the `change_return_type` peer for flag narrowing.

## Master
`e82cd1a598858f80bfb2d882ec1d275413a82dae`

## Probe
`/workspace/fd-crt-noensure-override-probe` — live `WRONG=yes` for both cases (`out.txt`).

## Notes / distinct from
- **#1344** — explicit interface impl related-method discovery with **default** flags (CS0539/CS0535); different root (starting from explicit impl), not `updateImplementations=false`.
- **#1265** — partial method DeclaringSyntaxReferences / CS8817.
- **#1367** / **#1368** — same Ensure hole on reorder/add_parameter; not CRT.
- Unit test `ChangeReturnType_Column_FalseFlags_OnlyTouchesSelected` deliberately picks a **non-virtual** same-line sibling so false flags stay compiling — it does not cover the override/interface drift path proven here.
- Pref #1 ctor/attr remove/reorder default-flag path is exhausted; this is CRT flag-narrowing + missing Ensure.
